core/src/main/java/org/testcontainers/containers/wait/HttpWaitStrategy.java

```diff
@@ -114,6 +127,12 @@ protected void waitUntilReady() {
                                     connection.getResponseCode()));
                         }
 
+                        String responseBody = getResponseBody(connection);
```

Comment:
We could move the parsing of the response body into the if clause. I would prefer this, since it's only needed if a `responsePredicate` is set.

Reply:
Sure, that makes sense. I've made the change now.